compiler/src/dmd/expressionsem.d

```diff
@@ -2113,9 +2113,9 @@ private bool functionParameters(const ref Loc loc, Scope* sc,
                 if (!arg.isLvalue())
                 {
                     auto v = copyToTemp(STC.exptemp, "__rvalue", arg);
-                    Expression ev = new DeclarationExp(arg.loc, v);
-                    ev = new CommaExp(arg.loc, ev, new VarExp(arg.loc, v));
-                    arg = ev.expressionSemantic(sc);
+                    eprefix = Expression.combine(
+                        eprefix, (new DeclarationExp(arg.loc, v)).expressionSemantic(sc));
```

Comment:
Hmm, this looks like it messes with the argument expressions evaluation order - evaluating these `in` rvalues before all other argument expressions, regardless of their position.

Reply:
Good catch! Changed the fix and added a test.